### DISCUSSION TITLE
One of the constraints evaluates to a FALSE LOGICAL variable

### DISCUSSION THREAD

**Discussion by studydog-pro** [yalmip/YALMIP#1402] [Optimization modelling] [answered]
Dear Johan, I encountered this error prompt while editing constraint conditions. May I know the reason? How can I solve this problem?What should I do with the inverse matrix of "PDN" that I have in mind?

![1715651468105](https://github.com/yalmip/YALMIP/assets/150098588/b18555c4-3238-4336-983e-9bd8b6286cca)
![1715651487043](https://github.com/yalmip/YALMIP/assets/150098588/1b44b046-dfb2-4db2-9d91-5d91b45e1f75)

**Comment by johanlofberg** [comment, marked as answer]
You're saying PDN*0 == 1, which trivially is impossible

https://yalmip.github.io/fixedconstraints

**Reply by studydog-pro** [reply]
Thank you for your help. But I set 'PDN_inv' to a variable of type sdpvar also shows this error. What should I do to get the inverse of a matrix, which is of type sdpvar?

**Reply by johanlofberg** [reply]
No it doesn't

```
>> PDN = sdpvar(2,2,3,'full');
>> PDN_inv = sdpvar(2,2,3,'full');
>> CS = [];for i = 1:3;CS = [CS, PDN_inv(:,:,i)*PDN(:,:,i) == eye(2)];end;CS
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
|   ID|                           Constraint|   Coefficient range|
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
|   #1|   Equality constraint (bilinear) 2x2|              1 to 1|
|   #2|   Equality constraint (bilinear) 2x2|              1 to 1|
|   #3|   Equality constraint (bilinear) 2x2|              1 to 1|
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```

There is no good way to model inverse, unless you can perform a clever variable changes, congruence transformations, perform smart Schur complements etc

**Reply by studydog-pro** [reply]
Thanks for your patience in replying, I'll try other methods or change my model, thanks again!
